### ISSUE TITLE
How to strip CSS/style nodes?

### ISSUE THREAD

**Issue by GeeCastro** [matthewwithanm/python-markdownify#100]
```python
>>> from markdownify import markdownify as md
>>> md('<!DOCTYPE html><html><head><style>body {background-color: powderblue;}h1 {color: blue;}p {color: red;}</style></head><body><p>This is text</p></body></html>') 
'body {background-color: powderblue;}h1 {color: blue;}p {color: red;}This is text\n\n'
```
I may be misusing the option `strip` but I can't get rid of the style bits

**Comment by chrispy-snps** [comment]
@Chichilele - It looks like the implementation for `strip` is incomplete. It prevents markup generation for the containing element, but not for its contents:

```
>>> from markdownify import markdownify as md
>>> md('<pre>1</pre><pre>2</pre>')
'\n```\n1\n```\n\n```\n2\n```\n'
>>> md('<pre>1</pre><pre>2</pre>', strip=['pre'])
'12'

>>> md('<ol><li>a</li><li>b</li></ol>')
'1. a\n2. b\n'
>>> md('<ol><li>a</li><li>b</li></ol>', strip=['ol', 'li'])
'ab'
```

so I think you are just getting the text content of your `<style>` elements.

Perhaps this is the intended behavior; I'm not sure.

**Comment by tlk** [comment]
Another simple workaround:
```python
import re
my_html = '...'
stripped_html = re.sub(r'<style>(.*)</style>', '', my_html, flags=re.DOTALL)
markdown = md(stripped_html)
```

https://docs.python.org/3/library/re.html#re.DOTALL

**Comment by matthewwithanm** [comment]
Couldn't you just use a custom converter with a `convert_style` method, like shown in the [README][1]?

[1]: https://github.com/matthewwithanm/python-markdownify#creating-custom-converters

**Comment by matthewwithanm** [comment]
> Perhaps simply stripping the markup of a tag is the intended behavior, and we need a new `ignore` argument to completely ignore a tag _and_ its contents.

Yeah, it's not about stripping content, but stripping tags. Generally, the body of the elements is actual content being marked up; style is a bit of an outlier because the body isn't really content.

A new option makes sense. We could probably also do with a README update, though; I don't think people realize that you can write custom `convert_blah` functions instead of resorting to preprocessing the input.

**Comment by tlk** [comment]
@matthewwithanm  I suggest you apply a bugfix along the lines of what was suggested by @aytey.

I read the readme section about custom converters and it starts with "If you have a special usecase". The thing is, I do not think this style-issue is a special use case at all - in fact, I cannot think of a use case where the current behaviour is desirable. Maybe I am missing something?

**Comment by matthewwithanm** [comment]
@tlk Yeah, I'm not saying the library shouldn't handle this.

All I'm saying is that, if you find a case where a tag isn't being handled like you want, you don't need a patch like @aytey's suggestion, or to preprocess like you suggested. You just need to subclass and add a `convert_style` method. And this should be made clearer in the README so people know they don't have to modify the library or do regex stuff.

**Comment by Chichilele** [comment]
I'm glad this opened a conversation. Although I went down another route for my project, here is what I would have used to solve this issue (based on the suggestions above):
```python
from markdownify import MarkdownConverter

class MarkdownConverterNoStyle(MarkdownConverter):
    def convert_style(self, el, text, convert_as_inline):
        return ''

html = '<!DOCTYPE html><html><head><style>body {background-color: powderblue;}h1 {color: blue;}p {color: red;}</style></head><body><p>This is text</p></body></html>'
MarkdownConverterNoStyle().convert(html)
```
Returns a nice `'This is text\n\n'`. 

The `convert_<tag name>` template wasn't obvious to me but is in fact very handy @matthewwithanm. Perhaps adding this example to the readme would be good? Or replacing the existing? 

@tlk I share your opinion that style should be left out by default

**Comment by tlk** [comment]
@matthewwithanm Got it - good point!

Come to think of it, the <kbd>script</kbd> tag would probably also need to be dealt with.

PR  https://github.com/matthewwithanm/python-markdownify/pull/112 will make the markdownify library ignore both the <kbd>script</kbd> tag and the <kbd>style</kbd> tag.

**Comment by chrispy-snps** [comment]
Fixed in 0.12.1.
